lib/src/Uncertainty/Algorithm/MetaModel/FunctionalChaos/FunctionalChaosResult.cxx

```diff
@@ -307,6 +313,101 @@ void FunctionalChaosResult::setInvolvesModelSelection(const Bool involvesModelSe
   involvesModelSelection_ = involvesModelSelection;
 }
 
+/* Conditional expectation accessor */
+FunctionalChaosResult FunctionalChaosResult::getConditionalExpectation(const Indices & conditioningIndices) const
+{
+  // Get marginal input sample and distribution
+  const UnsignedInteger inputDimension = inputSample_.getDimension();
+  if (!conditioningIndices.check(inputDimension))
+    throw InvalidArgumentException(HERE) << "The conditioning indices must be in the range [0, dim-1] and must be different.";
+  const Sample inputSampleMarginal(inputSample_.getMarginal(conditioningIndices));
+  const Distribution inputDistributionMarginal(distribution_.getMarginal(conditioningIndices));
+  
+  // Check independence
+  if (!distribution_.hasIndependentCopula())
+    throw InvalidArgumentException(HERE) << "FunctionalChaosResult can only compute the conditional expectation for an independent copula.";
+
+  // Create the conditioned orthogonal basis
+  if (!orthogonalBasis_.getImplementation()->isTensorProduct())
```

Comment:
The `isTensorProduct()` and `getMarginal()` method could easily be added to the `SoizeGanhemFactory` class. The first method has simply to return hasIndependentCopula_, and the second one has to return productPolynomial_.getMarginal(i)

Reply:
Thank you for pointing this: otherwise, since SoizeGhanemFactory inherits from OrthogonalFunctionFactory, the isTensorProduct() method will always return False. This would be a bug, since The Ghanem-Soize factory can be a tensor product basis, provided the copula is independent, as you mentioned. This is fixed in 866fae63cea27aa92bce6cb85725119c6351803c